UCM-Powershell-IIS/Instalation.ps1

```diff
@@ -0,0 +1,336 @@
+   Remove-Item "$ConfigDirectory/temp.json"
+   Rename-Item "$ConfigDirectory/temp2.json" "$ConfigDirectory/dashboard.json"
+   Write-Host "Creating dashboard '$DashboardName' on Uptime Cloud Monitor interface for MS Web Server"
+   Create-Dashboard $ApiHost $UserAPIKey $service $DashboardName
+}
+
+function CreateXML
+{
+  Write-Host "Saving settings"
+  $XMLWriter = New-Object System.XMl.XmlTextWriter($XmlfilePath, $Null)
+
+  $XMLWriter.Formatting = "Indented"
+  $XMLWriter.Indentation = "4"
+
+  $XMLWriter.WriteStartDocument()
+
+  $XMLWriter.WriteStartElement("Settings")
+
+  $XMLWriter.WriteStartElement("UptimeCloudMonitor")
+  $XMLWriter.WriteElementString("Apikey",$script:UserAPIKey)
+  $XMLWriter.WriteElementString("ApiServer",$script:ApiHost)
+  $XMLWriter.WriteEndElement()
+
+  $XMLWriter.WriteStartElement("MetricGroups")
+
+  $server_count = $MetricGroup.Servers.Count
+  if($server_count -gt 0)
+  {
+    $XMLWriter.WriteStartElement("MetricGroup$Count")
+    $XMLWriter.WriteElementString("ServiceName", $service)
+    $XMLWriter.WriteElementString("DashboardName", "$($MetricGroup.DashboardName)")
+    $XMLWriter.WriteElementString("MetricGroupName", "$($MetricGroup.MetricGroupName)")
+    $XMLWriter.WriteElementString("MetricGroupLabel", "$($MetricGroup.MetricGroupLabel)")
+    $XMLWriter.WriteElementString("Frequency", "$($MetricGroup.Frequency)")
+
+    $XMLWriter.WriteStartElement("Servers")
+    $Index = 1
+
+    ForEach ($Server in $MetricGroup.Servers)
+    {
+      $XMLWriter.WriteStartElement("Server$Index")
+      $XMLWriter.WriteElementString("InstanceName",$Server.InstanceName)
+      $XMLWriter.WriteElementString("Hostname",$Server.Hostname)
+      $XMLWriter.WriteElementString("Username",$Server.Username)
+      $XMLWriter.WriteElementString("Password",$Server.Password)
+      $XMLWriter.WriteElementString("SystemIdentifier",$Server.UniqueName)
+      # Closing particular server, say Server1
+      $XMLWriter.WriteEndElement()
+      $Index++
+    }
+    # Close Servers tag
+    $XMLWriter.WriteEndElement()
+
+
+    # Close a particular MetricGroup tag, say MetricGroup1 tag
+    $XMLWriter.WriteEndElement()
+    $Count++
+  }
+
+  # Close MetricGroup
+  $XMLWriter.WriteEndElement()
+
+  # Close Settings
+  $XMLWriter.WriteEndElement()
+
+  # End the XML Document
+  $XMLWriter.WriteEndDocument()
+
+  # Finish The Document
+  $XMLWriter.Flush()
+  $XMLWriter.Close()
+}
+
+function ConfigureInstanceSpecificDetails($metric, $InstanceNumber)
+{
+  Write-Host "Unique name for this instance ? [Default = Instance$InstanceNumber]"
+  PrintHintText $script:UniqueNameHintText
+  $UniqueName = Read-Host
+  if ($UniqueName -eq "")
+  {
+    $UniqueName = "Instance$InstanceNumber"
+  }
+
+  Write-Host "Servername/Hostname ?"
+  $InstanceName = Read-Host
+  if ($InstanceName -eq "")
+  {
+    Write-Host "Continuing with blank instance name. We will set if to localhost"
+  }
+}
+
+function ConfigureSQLServerAuthentication
```

Comment:
rename methods. Also remove username and password in IIS if not required. This goes for code, config file, installation script and everything else.